File: algorithms/python/937/main.py
class Solution:
    def reorderLogFiles(self, logs: List[str]) -> List[str]:
        letters, digit = [], []
        logs = list(reversed(logs))
        while logs:
            log = logs.pop()
            if (log.split()[1]).isdigit():
                digit.append(log)
            else:
                letters.append(log)
        letters.sort(key=lambda x: (x.split()[1:], x.split()[0]))
        return letters + digit

File: algorithms/cpp/937/main.cpp
#include <vector>
#include <utility>
#include <iostream>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<string> reorderLogFiles(vector<string>& logs) {
        vector<string> letters;
        vector<string> res;
        vector<pair<string, string>> digit;
        for(string log: logs){
            int idx = log.find(' ') + 1;
            if(log[idx] >= '0' && log[idx] <='9'){
                letters.push_back(log);
            } else{
                digit.push_back(make_pair(log.substr(idx), log.substr(0, idx)));
            }
            cout << log;
        }
        sort(digit.begin(), digit.end());
        for(pair<string, string> log: digit){
            res.push_back(log.second + log.first);
        }
        for(string log: letters){
            res.push_back(log);
        }
        return res;
    }
};